pkg/master/master_test.go

```diff
@@ -363,6 +378,107 @@ func TestAPIVersionOfDiscoveryEndpoints(t *testing.T) {
 
 }
 
+// This test doesn't cover the apiregistration and apiextensions group, as they are installed by other apiservers.
+func TestStorageVersionHashes(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StorageVersionHash, true)()
+	master, etcdserver, _, _ := newMaster(t)
+	defer etcdserver.Terminate(t)
+
+	server := httptest.NewServer(master.GenericAPIServer.Handler.GoRestfulContainer.ServeMux)
+
+	c := &restclient.Config{
+		Host:          server.URL,
+		APIPath:       "/api",
+		ContentConfig: restclient.ContentConfig{NegotiatedSerializer: legacyscheme.Codecs},
+	}
+	discover := discovery.NewDiscoveryClientForConfigOrDie(c)
+	all, err := discover.ServerResources()
+	if err != nil {
+		t.Error(err)
+	}
+	for _, g := range all {
```

Comment:
probably add another check that the number of entries in 'all' equals the number of entries in storageversionhashdata.

Reply:
I don't think that check adds value. We have already checked that the entries in the discovery doc is either a subresource, or in the 
storageversionhashdata#NoStorageVersionHash or 
in the storageversionhashdata#GVRToStorageVersionHash.

Reply:
the inverse is not checked, i think. some entries in storageversionhashdata left over while their APIs have been removed.

Reply:
Good point. Added.